lib/Email/MIME/Encode.pm

```diff
@@ -16,6 +17,10 @@ sub maybe_mime_encode_header {
     $header = lc $header;
 
     my $header_length = length($header) + length(": ");
+
+    return $val->as_mime_string($charset, $header_length)
```

Comment:
Okay, I'm sold.  Let's reverse the argument order so that it's easier to give the header length and let the encoder pick an encoding, if the client doesn't care?